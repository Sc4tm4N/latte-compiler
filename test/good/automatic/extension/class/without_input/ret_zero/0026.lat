class a1 {
  int i;
  string s;
  boolean b;
  a2 a;

  void change(a2 a, int howMuch) {
    // what can go wrong?
    a.i = a.i + 1; // 100, 111111
    a.s = a.s + "-->a1";
    a.b = !a.b;
    if(howMuch <= 0)
      return;
    else
      a.change(self, howMuch - 1);
  }
}

class a2 {
  int i;
  string s;
  boolean b;
  a1 a;

  void change(a1 a, int howMuch) {
    // what can go wrong?
    a.i = a.i + 1;
    a.s = a.s + "-->a2";
    a.b = !a.b;
    if(howMuch <= 0)
      return;
    else
      a.change(self, howMuch - 1);
  }
}

int main() {
  a1 a1 = new a1;
  a1.i = 100;
  a1.s = "Init a1! ||";
  a1.b = true;

  a2 a2 = new a2;
  a2.i = 10;
  a2.s = "Init a2! ||";
  a2.b = false;

  a2.a = a1;
  a1.a = a2;

  a1.change(a2, 10);

  printInt(a1.i);
  printString(a1.s);
  printBoolean(a1.b);

  printInt(a2.i);
  printString(a2.s);
  printBoolean(a2.b);

  return 0;
}

void printBoolean(boolean b) {
  if(b) {
    printString("true");
  } else {
    printString("false");
  }
}