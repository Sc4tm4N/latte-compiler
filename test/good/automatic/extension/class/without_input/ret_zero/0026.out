105
Init a1! ||-->a2-->a2-->a2-->a2-->a2
false
16
Init a2! ||-->a1-->a1-->a1-->a1-->a1-->a1
false
